### PR TITLE
Add report of tcp connection open event

### LINKED ISSUE #1835: Open connection.event will not be reported
Though three connection events are added by https://github.com/istio/proxy/pull/1371, connection `open` event will never be reported.

With current logic `open` is recorded when check is called for a TCP connection. However report won't be called until either periodic report triggered or connection closing, which rewrites connection event to either `continue` or `close`. 

Should we make a report call at onNewConnection?

@qiwzhang @JimmyCYJ

### PR DESCRIPTION
**Pull request by bianpengyuan**
Add a report call after tcp check finishes and before report timer starts, which record tcp connection open event. fix #1835

### CODE REVIEW

**Comment by bianpengyuan** [comment]
@qiwzhang

File: src/istio/control/tcp/request_handler_impl.cc [hunk 1]
```diff
@@ -55,16 +55,17 @@ CancelFunc RequestHandlerImpl::Check(CheckData* check_data,
 
 // Make remote report call.
 void RequestHandlerImpl::Report(ReportData* report_data) {
```
Comment (qiwzhang) [inline src/istio/control/tcp/request_handler_impl.cc]: I think we can remove this function. It was there when mixer_client library were at different repo.

**Comment by bianpengyuan** [comment]
Ideally we would like to report open connection OnNewConnection, but it won't have principal information in auth case. So report open connection when check is ok.

@qiwzhang Let me know if you want me to make that boolean args into enum to improve readability.

**Comment by qiwzhang** [comment]
Yeap,  I think making it to enum is better for readability.  We only need:  first,  intermediate, and last report, is that right?

**Comment by bianpengyuan** [comment]
@qiwzhang Done, ptal.

File: src/istio/control/tcp/request_handler_impl.cc (hunk 1 above)
Comment (bianpengyuan) [reply src/istio/control/tcp/request_handler_impl.cc]: Sure, removed.

**Comment by bianpengyuan** [comment]
@hklai Can you help to merge this? Connection.event is documented as having three possible values while right now connection open event is not even reported. This PR fixes the bug. The risk is low. It has no thing to do with upgrading, no new attributes type and only one report call added to async Report buffer.

@douglas-reid @gargnupur

**Pull request merged**